code/game/machinery/doors/airlock.dm

```diff
@@ -1175,8 +1176,17 @@
 	delayed_close_requested = FALSE
 	if(safe)
 		CheckForMobs()
+	catchHair()
 	return TRUE
 
+/obj/machinery/door/airlock/proc/catchHair()
+	if(prob(10))
+		for(var/mob/living/carbon/human/H in orange(2, src))
+			if(H.hair_style == "Braid (Floorlength)")
```

Comment:
The problem is hairstyles are stringly typed.  I suppose I could re-factor hairstyles.

Reply:
I looked into refactoring hair to not be stringly typed.

Way more work than I'm willing to put in for a meme PR.

If hairstyle is just a string, not an object or datum, how should I do this?

Reply:
String defined hairstyles(and other visuals like beards) are bad in every possible way and there isnt a way to polish a turf
So yeah apart from datumizing them(which isnt hard at all actually, just kinda tedious copypaste) there is not a way to make it work nicely
At best i suggest adding a catchable component to a human who aquired the braids hairstyle, at least it will be like quarter of the proper solution(addition of catchable component)